### PR TITLE
ensure search on collapsed nodes

### PR DESCRIPTION
**Pull request by SYBIOTE**
## References
closes #_insert number here_

### CODE REVIEW

File: packages/editor/src/panels/hierarchy/helpers.ts [hunk 1]
```diff
@@ -128,25 +139,32 @@ function buildHierarchyTree(
...
     isLeaf: !(node.children && node.children.length > 0),
-    lastChild: lastChild
+    lastChild: lastChild,
+    isRendered: isRendered
   }
   array.push(item)
 
-  if (hasComponent(entity, GLTFComponent) && showModelChildren) {
-    const scene = GLTFComponent.getInstanceID(entity)
+  if (hasComponent(entity, ModelComponent) && showModelChildren) {
```
Comment (HexaField) [inline packages/editor/src/panels/hierarchy/helpers.ts]: please revert. we no longer use the ModelComponent, it is deprecated and only exists for comparison.

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/editor/src/panels/hierarchy/helpers.ts (hunk 1 above)
Comment (SYBIOTE) [reply packages/editor/src/panels/hierarchy/helpers.ts]: roger that

Comment (SYBIOTE) [reply packages/editor/src/panels/hierarchy/helpers.ts]: done

File: packages/editor/src/panels/hierarchy/helpers.ts [hunk 2]
```diff
@@ -128,25 +139,32 @@ function buildHierarchyTree(
...
     isLeaf: !(node.children && node.children.length > 0),
-    lastChild: lastChild
+    lastChild: lastChild,
+    isRendered: isRendered
   }
   array.push(item)
 
   if (hasComponent(entity, GLTFComponent) && showModelChildren) {
-    const scene = GLTFComponent.getInstanceID(entity)
+    const modelSceneID = getModelSceneID(entity)
```
Comment (HexaField) [inline packages/editor/src/panels/hierarchy/helpers.ts]: need to also revert the usage of getModelSceneID, it is also not used now.

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/editor/src/panels/hierarchy/helpers.ts (hunk 2 above)
Comment (SYBIOTE) [reply packages/editor/src/panels/hierarchy/helpers.ts]: updating

**Review by HexaField — APPROVED** [review]

**Pull request merged**